app/src/main/java/net/gsantner/opoc/ui/FileSearchDialog.java

```diff
@@ -0,0 +1,317 @@
+                        }
+                        return true;
+                    }
+                    return false;
+                });
+
+                LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+                int px = (int) (new ContextUtils(searchEditText.getContext()).convertDpToPx(8));
+                lp.setMargins(px, px / 2, px, px / 2);
+
+
+                dialogLayout.addView(searchEditText, lp);
+                return searchEditText;
+            }
+
+
+            private static class OptionsMenuLayout {
+
+                private static LinearLayout init(final Initializer initializer, final LinearLayout dialogLayout) {
+                    final LinearLayout menuLayout = new LinearLayout(initializer._activity);
+                    menuLayout.setOrientation(LinearLayout.HORIZONTAL);
+                    menuLayout.setGravity(Gravity.CENTER);
+
+                    initMenuItems(initializer, menuLayout);
+
+                    dialogLayout.addView(menuLayout);
+
+                    return menuLayout;
+                }
+
+
+                private static LinearLayout.LayoutParams getLayoutParams() {
+                    final int leftMargin = 40;
+                    final int rightMargin = 40;
+                    final int topMargin = 40;
+                    final int bottomMargin = 40;
+                    LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+                    layoutParams.setMargins(leftMargin, topMargin, rightMargin, bottomMargin);
+
+                    return layoutParams;
+                }
+
+
+                private static void initMenuItems(final Initializer initializer, final LinearLayout menuLayout) {
+                    LinearLayout.LayoutParams layoutParams = getLayoutParams();
+
+                    ImageView imgRegex = initRegexOption(initializer);
+                    ImageView imgCaseSensitive = initCaseSensitivityOption(initializer);
+                    ImageView imgSearchInContent = initSearchInContentOption(initializer);
+
+                    menuLayout.addView(imgRegex, layoutParams);
+                    menuLayout.addView(imgCaseSensitive, layoutParams);
+                    menuLayout.addView(imgSearchInContent, layoutParams);
+                }
+
+
+                private static ImageView initRegexOption(final Initializer initializer) {
+                    final Options.SearchConfigOptions searchConfig = initializer._dialogOptions.searchConfigOptions;
+
+                    ImageView imgRegex = new ImageView(initializer._activity);
+                    imgRegex.setImageResource(R.drawable.ic_regex_black_24dp);
+                    updateRegexOption(imgRegex, searchConfig.isRegexQuery, initializer._activity);
```

Comment:
When I first looked at it I was not at first sight sure what does what.

Could you please use a simple CheckBox instead of a ImageView? So checkbox item with description text set.

example from drawing tool:

![grafik](https://user-images.githubusercontent.com/6735650/113922785-0363a180-97e8-11eb-99ad-44c2239ef147.png)

Reply:
> 
> 
> When I first looked at it I was not at first sight sure what does what.
> 
> So checkbox item with description text set.
> 

it takes too much screen space required, is any alternative?

Reply:
![image](https://user-images.githubusercontent.com/56083544/113985675-16608b00-9855-11eb-9b3a-7993c865cc20.png)
![image](https://user-images.githubusercontent.com/56083544/113985703-1e202f80-9855-11eb-85c5-af1cafa3c42c.png)

Reply:
I don't think it should be an issue to have 3 options vertically.

![grafik](https://user-images.githubusercontent.com/6735650/113989605-e57e5700-9850-11eb-99f4-8de90f8f5909.png)

It probably will be similar to 3 rows of filemanager rows combined in height.

Reply:
something like this?
![image](https://user-images.githubusercontent.com/56083544/113994033-b5898080-985d-11eb-9e89-074479001703.png)